src/SprykerEco/Yves/Payone/Handler/PayoneHandler.php

```diff
@@ -157,6 +159,12 @@ protected function setPayonePayment(Request $request, QuoteTransfer $quoteTransf
             $paymentDetailTransfer->setBankGroupType($payonePaymentTransfer->getBankGroupType());
             $paymentDetailTransfer->setIban($payonePaymentTransfer->getIban());
             $paymentDetailTransfer->setBic($payonePaymentTransfer->getBic());
+        } elseif ($paymentSelection == PaymentTransfer::PAYONE_CASH_ON_DELIVERY) {
+            $shippingProvider = $quoteTransfer->getShipment()->getMethod()->getCarrierName();
+            $payonePaymentTransfer = $this->getPayonePaymentTransfer($quoteTransfer, $paymentSelection);
+
+            $payonePaymentTransfer->setShippingProvider($shippingProvider);
+            $paymentDetailTransfer->setShippingProvider($shippingProvider);
```

Comment:
Why do you need this twice?

Reply:
Removed unnecessary one `$payonePaymentTransfer->setShippingProvider($shippingProvider);`